### PR TITLE
async setContentsScheduled

### PR DESCRIPTION
**Pull request by hoebbelsB**
# Description

Since we are having troubles with initializing huge documents, the idea was to introduce a batching mechanism to process large deltas in chunks rather than doing everything sync. Please take a look at this clicket for further information about the idea and implementation details: https://staging.clickup.com/t/333/CLK-24538

### CODE REVIEW

File: core/editor.js [hunk 1]
```diff
@@ -11,11 +11,157 @@ import Header from '../formats/header';
 import TextBlot, { escapeText } from '../blots/text';
 
 const ASCII = /^[ -~]*$/;
+let postMessage = cb => {
```
Comment (volser) [inline core/editor.js]: can we move all scheduler stuff into separate file?

File: core/editor.js [hunk 2]
```diff
@@ -11,11 +11,157 @@ import Header from '../formats/header';
...
+};
 
 class Editor {
   constructor(scroll) {
     this.scroll = scroll;
     this.delta = this.getDelta();
+    this.scheduler = createScheduler();
+  }
+
+  async scheduledApplyDelta(delta, { signal, onProgress }) {
```
Comment (volser) [inline core/editor.js:23]: can we add some unit tests for these methods?

File: core/editor.js (hunk 1 above)
Comment (hoebbelsB) [reply core/editor.js]: done

File: core/editor.js (hunk 2 above)
Comment (hoebbelsB) [reply core/editor.js:23]: done

**Review by volser — APPROVED** [review]
LGTM

**Pull request merged**